### ISSUE TITLE
fm328.f  TODO: missing intrinsic lowering: iabs

### ISSUE THREAD

**Issue by schweitzpgi** [flang-compiler/f18-llvm-project#257]

**Comment by Sameeranjoshi** [comment]
#265 implements a part of it.
Following [TODO](https://github.com/flang-compiler/f18-llvm-project/blob/facf5f130108bb2e43c7f55402814f9f2a182bb6/flang/lib/Lower/ConvertExpr.cpp#L353) is failing to compile fm328.f now.

**Comment by Sameeranjoshi** [comment]
Should it be retitled now?

**Comment by jeanPerier** [comment]
I looked a bit into the issue, its a bit broader than just iabs. I overlooked a bit intrinsic procedure passed as arguments when they have a different name as the generic version.
Intrinsic lowering should only have the implementation of the generic intrinsic to avoid duplicating the work.
For intrinsic calls, the front-end is rewriting the specific to the generic because that is legal Fortran. It cannot do the same when passing a specific intrinsic (e.g `foo(iabs)` cannot be rewritten to `foo(abs)`). So, when lowering intrinsic passed as dummy, we need to retrieve generic name and use that to generate the code.

The patch I have in mind is small, so I will directly push here + llvm master (it impacts the front-end more than lowering), sorry to steal this from you Sameeran.
